### ISSUE TITLE
Better Routing front-end

### ISSUE THREAD

**Issue by martin-mito** [openstreetmap-ng/openstreetmap-ng#139]
The look of the routing menu is actually quite bad. The drop down menu with different routing providers is confusing and there is a lot more potential:
* Add via points
* More profiles (e.g. a hiking profile that allows routing via SAC scale >T2 or a racing or mountainbikes profile for bikers).
* Adding an altitude profile

I think all routing providers would allow these things, at least Graphhopper and Vallhalla do. Afaik OSRM is generally not that actively maintained at the moment?

I don't have much time at the moment, so I will donate some money to this great project and maybe it will be an further incentive to work on this issue. I guess Librepay is the best option for you?

**Comment by Zaczero** [comment]
I completely agree, and comparing OSM to other maps - we are lacking. I think, ideally, we should provide semi-feature-rich navigation controls (the ones that you described, covering 99% of the use cases), but also promote OSM-based routers (e.g. brouter) and link to them. So people can discover application-specific interfaces, and incentivise further development. We will never be able to cover 100% of the use cases so we may as well promote others :smiley:.

When talking about the timelines, this feature will likely be worked on after all of the groundwork for OSM-NG is finished. By the end of this year, we strive for the first testing release of the site, and then we will develop the remaining minor functionalities. After all of that, we can focus on significant improvements to the site (photon, better routing, etc.). Priorities first :wink:.

> I don't have much time at the moment, so I will donate some money to this great project and maybe it will be an further incentive to work on this issue. I guess Librepay is the best option for you?

Liberapay has a special place in my heart because it's open source. But both platforms (liberapay, gh sponsors) are completely fine with me so simply pick which you prefer. Also, I can't make any promises when this feature will be developed. I try to focus my work where it matters the most at the given point in time. Those features are nice-to-have but not critical.
